src/gs-update-monitor.c

```diff
@@ -150,6 +192,269 @@ no_updates_for_a_week (GsUpdateMonitor *monitor)
 	return FALSE;
 }
 
+static void
+app_set_auto_updating (GsApp *app,
+		       gboolean auto_updating)
+{
+	/* we always have to set it to NULL as otherwise GsApp doesn't allow
+	 * the metadata to be overridden */
```

Comment:
This need to always reset to `NULL` before setting anything seems like a flaw in `GsApp`. Worth fixing?

Reply:
I've tried to change these little safeguard upstream but Richard thinks they're a good thing... So this is me complying with that.